dask_expr/io/io.py

```diff
@@ -181,6 +183,168 @@ def _tune_up(self, parent):
+        return make_meta(meta)
+
+    def _divisions(self):
+        if self.operand("user_divisions"):
+            return self.operand("user_divisions")
+        else:
+            npartitions = len(self.iterables[0])
+            return (None,) * (npartitions + 1)
+
+    @property
+    def apply_func(self):
+        if self.enforce_metadata:
+            return apply_and_enforce
+        return self.func
+
+    @functools.cached_property
+    def apply_kwargs(self):
+        kwargs = self.kwargs
+        if self.enforce_metadata:
+            kwargs = kwargs.copy()
+            kwargs.update(
+                {
+                    "_func": self.func,
+                    "_meta": self._meta,
+                }
+            )
+        return kwargs
+
+    def _filtered_task(self, index: int):
+        vals = [v[index] for v in self.iterables]
+        if self.apply_kwargs:
+            return (apply, self.apply_func, vals + self.args, self.apply_kwargs)
+        else:
+            return (self.func, *vals, *self.args)
+
+
+class FromMapProjectable(FromMap):
+    _parameters = [
+        "func",
+        "iterables",
+        "columns",
+        "args",
+        "kwargs",
+        "user_meta",
+        "enforce_metadata",
+        "user_divisions",
+        "label",
+        "_partitions",
+        "_series",
+    ]
+    _defaults = {
+        "user_meta": no_default,
+        "enforce_metadata": False,
+        "user_divisions": None,
+        "label": None,
+        "_partitions": None,
+        "_series": False,
+    }
+    _absorb_projections = True
+
+    @functools.cached_property
+    def columns_operand(self):
+        return _convert_to_list(self.operand("columns"))
+
+    @property
+    def columns(self):
+        if self.columns_operand is None:
+            return list(self.frame_meta.columns)
+        else:
+            return self.columns_operand
+
+    @functools.cached_property
+    def _series(self):
+        # Only need to convert to _series if func
+        # doesn't produce a Series already
+        return self.operand("_series") and self.frame_meta.ndim > 1
+
+    @functools.cached_property
+    def kwargs(self):
+        options = self.operand("kwargs")
+        if self.columns_operand:
+            options["columns"] = self.columns_operand
+        return options
+
+    @functools.cached_property
+    def apply_kwargs(self):
+        kwargs = self.kwargs
+        if self.enforce_metadata:
+            kwargs = kwargs.copy()
+            kwargs.update(
+                {
+                    "_func": self.func,
+                    "_meta": self.frame_meta,
+                }
+            )
+        return kwargs
+
+    @functools.cached_property
+    def frame_meta(self):
+        # This is our `_meta` result before possibly
+        # converting to a Series
+        meta = super()._meta
+        if meta.ndim > 1 and self.columns_operand is not None:
+            return meta[self.columns_operand]
```

Comment:
This is confusing, shouldn't we use ``self.operand("columns")``? ``columns_operand`` is always a list, so we never coerce to a Series

Reply:
Ah sorry, forget it, missed meta below